sui-execution/latest/sui-move-natives/src/crypto/attestation.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_binary_format::errors::PartialVMResult;
+use move_core_types::gas_algebra::InternalGas;
+use move_vm_runtime::native_functions::NativeContext;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    natives::function::NativeResult,
+    pop_arg,
+    values::{Value, VectorRef},
+};
+use smallvec::smallvec;
+use std::collections::VecDeque;
+use sui_types::attestation::attestation_verify_inner;
+pub fn nitro_attestation_verify_inner(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 6);
+
+    // todo: figure out cost
+
+    let timestamp = pop_arg!(args, u64);
+    let pcr2 = pop_arg!(args, VectorRef);
+    let pcr1 = pop_arg!(args, VectorRef);
+    let pcr0 = pop_arg!(args, VectorRef);
```

Comment:
For future proofing, i think pcr register might change a bit based on the setup such as Intel TDX or AMD-SEV
[TDX](https://dl.acm.org/doi/10.1145/3652597#:~:text=Taking%20Measurements.,Table%203.)

Reply:
note that this native function would be specific to aws nitro. new native functions will be introduced for other enclave format.